Model/src/GamePackage/Player.java

```diff
@@ -1,4 +1,265 @@
+	 *****************************************************************/
+	public void setBankrupt(boolean isBankrupt) {
+		this.isBankrupt = isBankrupt;
+	}
+
+	/******************************************************************
+	 * Getter method for wallet variable.
+	 * 
+	 * @return wallet Returns the int wallet.
+	 *****************************************************************/
+	public int getWallet() {
+		return wallet;
+	}
+
+	/******************************************************************
+	 * Setter method for wallet variable.
+	 * 
+	 * @param wallet Sets the int wallet.
+	 *****************************************************************/
+	public void setWallet(int wallet) {
+		this.wallet = wallet;
+	}
+
+	/******************************************************************
+	 * Getter method for propertiesOwned ArrayList.
+	 * 
+	 * @return propertiesOwned Returns the ArrayList propertiesOwned.
+	 *****************************************************************/
+	public ArrayList<BoardSquare> getPropertiesOwned() {
+		return propertiesOwned;
+	}
+
+	/******************************************************************
+	 * Setter method for propertiesOwned ArrayList.
+	 * 
+	 * @param propertiesOwned Sets the ArrayList propertiesOwned.
+	 *****************************************************************/
+	public void setPropertiesOwned(ArrayList<BoardSquare> 
+	propertiesOwned) {
+		this.propertiesOwned = propertiesOwned;
+	}
+
+	/******************************************************************
+	 * Getter method for cardsHeld ArrayList.
+	 * 
+	 * @return cardsHeld Returns the ArrayList cardsHeld.
+	 *****************************************************************/
+	public ArrayList<Card> getCardsHeld() {
+		return cardsHeld;
+	}
+
+	/******************************************************************
+	 * Setter method for cardsHeld ArrayList.
+	 * 
+	 * @param cardsHeld Sets the ArrayList cardsHeld.
+	 *****************************************************************/
+	public void setCardsHeld(ArrayList<Card> cardsHeld) {
+		this.cardsHeld = cardsHeld;
+	}
+	
+	/******************************************************************
+	 * Checks to make sure player has sufficient funds before
+	 * subtracting amount from wallet. If the player has sufficient
+	 * funds the method subtracts amount from wallet and returns the
+	 * amount paid. If player doesn't have sufficient funds he method
+	 * subtracts what is able to be paid from wallet and returns the
+	 * amount that has been paid.
+	 * 
+	 * @param amount The amount of money requested from player.
+	 * @return amountPaid The amount of money that was paid by player.
+	 *****************************************************************/
+	public int pay(int amount) {
+		int amountPaid = 0;
+		
+		if(wallet >= amount) {
+			wallet -= amount;
+			amountPaid = amount;
+		}
+		else {
+			amountPaid = amount - wallet;
+			wallet -= amountPaid;
+		}
+		
```

Comment:
Look over the logic for the else. I think it should be amountPaid = wallet and then wallet -= amountPaid, which can be simply wallet = 0. Right? If amount desired is $30, but player only has $10, we dont want amountPaid to be $20, we want to pay all the player has which is $10 and set wallet to 0.

Reply:
Good catch you are right! Sorry about that.